File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/dto/response/FullHabitLogsForWeek.kt
package com.habit.habit_tracker.dto.response

import com.fasterxml.jackson.annotation.JsonInclude
import com.habit.habit_tracker.dto.response.HabitResponse

@JsonInclude(JsonInclude.Include.NON_NULL)
data class FullHabitLogsForWeek(
    val dailyHabitLogs: List<DailyHabitLogResponse>,
    val weeklyHabitLog: WeeklyHabitLogResponse?,
    val habit: HabitResponse,
)

File: client/src/types/dto/response/FullHabitLogsForWeek.ts
import { DailyHabitLogResponse } from "./DailyHabitLogResponse";
import { WeeklyHabitLogResponse } from "./WeeklyHabitLogResponse";
import { HabitResponse } from "./HabitResponse";

export interface FullHabitLogsForWeek {
  dailyHabitLogs: DailyHabitLogResponse[];
  weeklyHabitLog?: WeeklyHabitLogResponse | null;
  habit: HabitResponse;
}
